lib/mapUtils.js

```diff
@@ -292,3 +302,70 @@ export function openDirections(latitude, longitude, storeName) {
     alwaysIncludeGoogle: true,
   });
 }
+
+// Find the distance between 2 points
+export function findDistance(pointA, pointB) {
+  const distanceMeters = getDistance(pointA, pointB);
+  // Convert distance to 'x.xx' form, in miles units
+  const distance = convertDistance(distanceMeters, 'mi').toFixed(2);
+  return distance;
+}
+
+// TODO: @wangannie clean up
+export async function toggleFavoriteStore(navigation, storeId) {
+  const customerId = await AsyncStorage.getItem('customerId');
+  const guest = customerId === RecordIds.guestCustomerId;
+  let success = false;
+  if (guest) {
+    Alert.alert(
+      'Account Required',
+      'Log in or create an account to favorite stores and get alerts about new deliveries',
+      [
+        {
+          text: 'Log in or create an account',
+          onPress: async () => completeLogout(navigation, true),
+        },
+        { text: 'Not now', onPress: async () => null },
+      ]
+    );
+  } else {
+    const cust = await getCustomersById(customerId);
+    let favoriteStores = cust.favoriteStoreIds || [];
+
+    if (favoriteStores.includes(storeId)) {
+      // Un-favorite a store
+      favoriteStores = favoriteStores.filter((store) => store !== storeId);
+    } else {
+      // Favorite a store
+      favoriteStores.push(storeId);
+      if (!cust.notifications) {
+        Alert.alert(
+          'Get updates on your favorite stores',
+          'Enable notifications to be alerted when your favorite stores receive deliveries!',
+          [
+            { text: 'Not now', style: 'cancel', onPress: async () => null },
+            {
+              text: 'Enable notifications',
+              style: 'default',
+              onPress: async () => navigation.navigate('Settings'), // TODO: link to specific page in settings
+            },
+          ]
+        );
+      }
+    }
+    await updateCustomers(customerId, {
+      favoriteStoreIds: favoriteStores,
+    });
+    success = true;
+  }
+  return success;
+}
+
+// Checks if a store is one of the customer's favorites
+// TODO @wangannie rename
+export async function isFavoritefromCustomer(storeId) {
```

Comment:
Not sure if this is an efficient way to check if a store is one of the customer's favorites. I thought it wouldn't make sense to add as a boolean to the `stores` list since that's loaded when the app opens, and I thought it'd get messy updating that through favoriting/unfavoriting.